plugins/processors/regex/README.md

```diff
@@ -38,6 +40,19 @@ For tags transforms, if `append` is set to `true`, it will append the transforma
     pattern = ".*category=(\\w+).*"
     replacement = "${1}"
     result_key = "search_category"
+
+  # Replace metric element names such as "measurement", "tag" or "field" name
+  [[processors.regex.metrics]]
```

Comment:
This PR adds a new way to change measurement or tag or field names. We already had a way to change tag and field names. With this PR there are now two ways to change tag or field names, and one way to change measurement names. I think this is a little confusing.

Why not just add a way to change measurement name, so there is only one way to change each type? Does the new "metrics" way add functionality over the existing way?

Reply:
@reimda enlighten me how the existing way to rename tag and field names with regex is done?

Reply:
@reimda could you elaborate please?

Reply:
I think I must be misunderstanding something fundamental about this PR since you and Sven are happy with it :)

Before this PR, to transform a tag we added a processors.regex.tags subtable with a key set to the tag name we want to transform. To transform a field we added a processors.regex.fields subtable with a key set to the field name we want to transform.

This PR adds the ability to transform a measurement name. If we followed the same pattern as tags and fields, it seems like we would add a processors.regex.measurement subtable. Since there's only one measurement name, we don't need a key field.

Instead, this PR adds processors.regex.metrics. This is a subtable that can transform either a tag or a field or a measurement name. You choose which type with the key setting. This meets the requirement of adding a way to transform the measurement name, but also adds a new way to transform fields and tags.

I think for each processors.regex.metrics that transforms a tag, you could use a processors.regex.tags. For each processors.regex.metrics that transforms a field you could use a processors.regex.fields. Having two ways to do the same thing is going to cause confusion for our users.

Maybe a processors.regex.metrics transforming a tag can do more than a processors.regex.tags? I admit I didn't spend the time to understand every new unit test, but the docs don't say much. Is the difference that processors.regex.metrics can regex match on the names of the tags/fields/measurement and processors.regex.tags and processors.regex.fields can only match a static string?

Reply:
@reimda not sure I understand you point... :-( Without this PR you can change tag and field **values** but **not** their **name**. So we use `processors.regex.tags` and `processors.regex.fields` to change the tag and field **values**. The PR then provides `processors.regex.metrics` to change metric properties such as the measurement **name**, as well as tag **names** and field **names**. I named it `metrics` as it changes the metric properties, but I'm open for better names.

So my answer to 
> Is the difference that processors.regex.metrics can regex match on the names of the tags/fields/measurement and processors.regex.tags and processors.regex.fields can only match a static string?

is yes. :-) You can change the **names** using regexp, while currently you need to explicitly map the names. This can be quite cumbersome when getting some JSON or XML data with dozens of fields named e.g. `a_very_long_but_static_prefix_with_a_some_{{fieldname}}_0`... :-)

Reply:
Indeed, this PR adds ability to change tag or field **name** instead of their **value**.

@srebhan in my #9506 I proposed to use `processors.regex.tag_rename` and similar as name for the config. They do sound more clear to me what they intend to do. WDYT?

Reply:
Yeah I can do this, but it then makes the `key` option useless. If this is fine for you I can implement the change. @reimda what do you think? I agree that it is more clear. :-)